src/lib/ffi/ffi_rng.cpp

```diff
@@ -57,6 +57,93 @@ int botan_rng_init(botan_rng_t* rng_out, const char* rng_type)
       });
    }
 
+int botan_rng_init_custom(botan_rng_t* rng_out, const char* rng_name, void* context,
+                          int(* get_cb)(uint8_t* out, size_t out_len, void* context),
+                          int(* add_entropy_cb)(const uint8_t input[], size_t length, void* context))
+{
+return ffi_guard_thunk(__func__,[=]() -> int {
+   if(rng_out == nullptr)
```

Comment:
Should also be checking rng_name and the callback functions. (I suppose `context` can legitimately be NULL, if the functions themselves do not need it eg for calling a HW RNG)

Reply:
IMHO only `get_cb` should be mandatory, all others may be set to `NULL`, e.g., the strongswan RNG API does not offer adding entropy to the pool.